src/core/dune

```diff
@@ -14,15 +14,28 @@
                (:include ../ocaml-flags-standard.sexp)
                (:include ../ocaml-flags-configure.sexp)
                (:include ../ocaml-context-flags.sexp)))
-  (preprocess  (per_module
-                  ((action (run %{bin:cppo} -V OCAML:%{ocaml_version} %{input-file})) opamCompat opamSystem)
-                  ((action (run %{bin:cppo} %{read-lines:developer} %{input-file})) opamCoreConfig)
-                  ((action (run %{bin:cppo} -D "VERSION %{read-lines:version}" %{input-file})) opamVersion)))
   (wrapped     false))
 
+(rule
+  (with-stdin-from opamCompat.pp.ml
```

Comment:
since `pp` is conventionally used for ppx AFAIU, why did you choose `.pp.ml` here? I'd suggest to use for example `opamCompat.ml.in` instead.